precompiles/outposts/stride/types.go

```diff
@@ -94,8 +95,53 @@ func parseLiquidStakeArgs(args []interface{}) (common.Address, common.Address, *
 	return sender, token, amount, receiver, nil
 }
 
-// CreateMemo creates the memo for the StakeIBC actions - LiquidStake and Redeem.
-func CreateMemo(action, receiverAddress string) (string, error) {
+// parseRedeemStakeArgs parses the arguments from the Redeem Stake method call
+func parseRedeemStakeArgs(args []interface{}) (common.Address, common.Address, *big.Int, common.Address, string, error) {
+	if len(args) != 5 {
+		return common.Address{}, common.Address{}, nil, common.Address{}, "", fmt.Errorf(cmn.ErrInvalidNumberOfArgs, 5, len(args))
+	}
+
+	sender, ok := args[0].(common.Address)
+	if !ok {
+		return common.Address{}, common.Address{}, nil, common.Address{}, "", fmt.Errorf(cmn.ErrInvalidType, "sender", "", args[0])
+	}
+
+	token, ok := args[1].(common.Address)
+	if !ok {
+		return common.Address{}, common.Address{}, nil, common.Address{}, "", fmt.Errorf(cmn.ErrInvalidType, "token", "", args[1])
+	}
+
+	amount, ok := args[2].(*big.Int)
+	if !ok {
+		return common.Address{}, common.Address{}, nil, common.Address{}, "", fmt.Errorf(cmn.ErrInvalidType, "amount", "", args[2])
```

Comment:
The errors in this function are not correct I think. This should be:
`fmt.Errorf(cmn.ErrInvalidType, "amount", big.Int{}, args[2])`
You can use this as a reference: https://github.com/evmos/evmos/blob/53747598a1c7dfc9fc03b605b08c28c79d909383/precompiles/outposts/osmosis/types.go#L164